exporter/prometheusremotewriteexporter/helper.go

```diff
@@ -337,28 +337,27 @@ func addSingleIntHistogramDataPoint(pt *otlp.IntHistogramDataPoint, metric *otlp
 	countlabels := createLabelSet(pt.GetLabels(), nameStr, baseName+countStr)
 	addSample(tsMap, count, countlabels, metric)
 
-	// count for +Inf bound
-	var totalCount uint64
+	// cumulative count for conversion to cumulative histogram
+	var cumulativeCount uint64
 
 	// process each bound, ignore extra bucket values
 	for index, bound := range pt.GetExplicitBounds() {
 		if index >= len(pt.GetBucketCounts()) {
 			break
 		}
-		bk := pt.GetBucketCounts()[index]
+		cumulativeCount += pt.GetBucketCounts()[index]
 		bucket := &prompb.Sample{
-			Value:     float64(bk),
+			Value:     float64(cumulativeCount),
 			Timestamp: time,
 		}
```

Comment:
This can be preallocated:

buckets := make([]prompb.Sample, len(pt.GetBucketCounts()))
for ... {
  bucket := &buckets[index]
  bucket.Value = cumulativeCount
  bucket.Timestamp = time
}